### PR TITLE
feat: Migrate away from Gatsby to React + Vite

### PR DESCRIPTION
**Pull request by kasperkristensen**

### CODE REVIEW

**Review by pKorsholm — COMMENTED** [review]
Generally: This is so much faster 😍 

One thing: order details doesn't seem to be working, it requests `admin/orders/undefined` when accessing the details page

**Comment by patrick-medusajs** [comment]
Found a rendering issue with the filter pop-over.
Screen Shot 2022-11-18 at 11 40 42 AM

File: src/components/molecules/modal/layered-modal.tsx [hunk 1]
```diff
@@ -27,22 +27,20 @@ export type ILayeredModalContext = {
 
 const defaultContext: ILayeredModalContext = {
   screens: [],
-  push: (screen) => {},
-  pop: () => {},
-  reset: () => {},
+  push: (screen) => { },
+  pop: () => { },
+  reset: () => { },
```
Comment (patrick-medusajs) [inline src/components/molecules/modal/layered-modal.tsx:32]:
```suggestion
  push: (screen) => {},
  pop: () => {},
  reset: () => {},
```
ESLint error

File: src/components/molecules/modal/stepped-modal.tsx [hunk 2]
```diff
@@ -30,42 +30,36 @@ type ISteppedContext = {
...
-  submit: () => {},
-  reset: () => {},
-  setPage: (page) => {},
+  enableNextPage: () => { },
+  disableNextPage: () => { },
+  goToNextPage: () => { },
+  goToPreviousPage: () => { },
+  submit: () => { },
+  reset: () => { },
+  setPage: (page) => { },
```
Comment (patrick-medusajs) [inline src/components/molecules/modal/stepped-modal.tsx:39]:
```suggestion
  enableNextPage: () => {},
  disableNextPage: () => {},
  goToNextPage: () => {},
  goToPreviousPage: () => {},
  submit: () => {},
  reset: () => {},
  setPage: (page) => {},
```
ESLint error

File: src/components/molecules/search-bar.tsx [hunk 3]
```diff
@@ -22,13 +22,12 @@ const SearchBar: React.FC = () => {
...
-    return globalHistory.listen(({ action }) => {
-      if (action === "PUSH") {
-        closeModal()
-      }
-    })
-  }, [])
+    if (navigation.state === "submitting") {
+      closeModal()
+    }
+  }, [navigation.state])
```
Comment (patrick-medusajs) [inline src/components/molecules/search-bar.tsx:30]: Bug: The topbar search modal doesn't not close upon clicking on a search result.

**Review by patrick-medusajs — COMMENTED** [review]
Went through each file with a diff and manually tested the relevant feature. Only found an issue with the topbar search and the table filter dropdown. 
Nice work 🎉

**Comment by olivermrbl** [comment]
What a great improvement to devexp! LGTM 💪

**Comment by kasperkristensen** [comment]
> issue: Found a rendering bug with the filter pop-over.
> 
> Screen Shot 2022-11-18 at 11 40 42 AM

It was missing a z-index, should be good now 💪

**Comment by kasperkristensen** [comment]
> Generally: This is so much faster 😍
> 
> One thing: order details doesn't seem to be working, it requests `admin/orders/undefined` when accessing the details page

Should be good now

**Review by patrick-medusajs — APPROVED** [review]
I pulled the latest and the reported bugs have been fixed. 
Found a dangling console.log to remove.
LGTM 💪

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**